blocks/bricks/recurrent.py

```diff
@@ -169,17 +172,20 @@ def only_given(arg_names):
                 states_given[name] = tensor.unbroadcast(state,
                                                         *range(state.ndim))
 
-            # Apply methods
-            if not iterate:
-                return application_function(brick, **kwargs)
-
             def scan_function(*args):
                 args = list(args)
                 arg_names = (list(sequences_given) + list(states_given) +
                              list(contexts_given))
                 kwargs = dict(zip(arg_names, args))
                 kwargs.update(rest_kwargs)
-                return application_function(brick, **kwargs)
+                outputs = getattr(brick, application_function.__name__)(
```

Comment:
Why the `getattr(brick, application_function.__name__)` instead of `application_function(brick, ...)`? I find the first one clearer (and shorter), actually, but maybe that's just a matter of taste.